src/hooks/useDarkMode.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react';
+
+const isSystemDarkMode = (): boolean => {
+  if (typeof window === 'undefined' || !window.matchMedia) return false;
+  return window.matchMedia('(prefers-color-scheme: dark)').matches;
+};
+
+export const useDarkMode = (): boolean => {
+  const [isDarkMode, setIsDarkMode] = useState<boolean>(false);
+
+  useEffect(() => {
+    const isDark = isSystemDarkMode();
+    setIsDarkMode(isDark);
+    document.documentElement.setAttribute('data-theme', isDark ? 'dark' : 'light');
```

Comment:
You aren't referencing this anywhere in your CSS at the moment. You can remove if if you want.